core/LSU.cpp

```diff
@@ -122,6 +153,12 @@ namespace olympia
     // Receive new load/store instruction from Dispatch Unit
     void LSU::getInstsFromDispatch_(const InstPtr &inst_ptr)
     {
+        if(!instPresentInQueues(inst_ptr)){
```

Comment:
Is this condition ever going to be false? If this condition is false, then does it mean, dispatch sent an instruction which is already there in the ldst issue queue?

Reply:
Yes this condition can be false. Currently there is a callback that is registered in the same method.
One for one of operands and other for the data 
https://github.com/riscv-software-src/riscv-perf-model/pull/92/files/11626bef0bcdc815dd6cc9b37dbe7f2d6d558858#diff-f42bfd79bf07a70c3e64bbe0c640480fd5ee792c0c0465f138d9bad9219959a0R183

https://github.com/riscv-software-src/riscv-perf-model/pull/92/files/11626bef0bcdc815dd6cc9b37dbe7f2d6d558858#diff-f42bfd79bf07a70c3e64bbe0c640480fd5ee792c0c0465f138d9bad9219959a0R170